docs/data-definition.md

```diff
@@ -0,0 +1,75 @@
+!!! danger "This page is for use with [the OpenSAFELY Databuilder](https://github.com/opensafely-core/databuilder/) which is currently unreleased."
```

Comment:
thought: I haven't extensively read through this or know it's exactly what we want, but it's fine as a starting point, given this isn't "live" documentation.